src/Engine/EmitMSIL/Replication.cs

```diff
@@ -86,44 +87,14 @@ internal static List<object> UnmarshalFunctionArguments(List<CLRFunctionEndPoint
                 }
                 catch (InvalidOperationException)
                 {
-                    string message = String.Format(Resources.kFFIFailedToObtainObject, paramType.Name, formalParams[i].CLRInfo.Member.DeclaringType.Name, formalParams[i].CLRInfo.Member.Name);
+                    string message = String.Format(Resources.kFFIFailedToObtainObject, paramType.Name, formalParams[i].CLRType.DeclaringType.Name, formalParams[i].CLRType.Name);
                     runtimeCore.LogWarning(ProtoCore.Runtime.WarningID.AccessViolation, message);
                     return null;
                 }
             }
             return values;
         }
 
-        internal static List<CLRFunctionEndPoint> CreateFEPs(IEnumerable<MethodBase> methods)
-        {
-            List<CLRFunctionEndPoint> feps = new List<CLRFunctionEndPoint>();
-            foreach (var mInfo in methods)
-            {
-                List<CLRFunctionEndPoint.ParamInfo> formalParams = new List<CLRFunctionEndPoint.ParamInfo>();
-
-                if (!mInfo.IsStatic && !mInfo.IsConstructor)
-                {
-                    // First argument is the this pointer
-                    // So we need to add an extra parameter to match the arguments
-                    var thisPtrType = mInfo.DeclaringType;
-                    var dsType = ProtoFFI.CLRObjectMarshaler.GetProtoCoreType(thisPtrType);
-                    formalParams.Add(new CLRFunctionEndPoint.ParamInfo() { ThisPtrType = thisPtrType, ProtoInfo = dsType });
-                }
-                
-                foreach (var param in mInfo.GetParameters())
-                {
-                    var dsType = ProtoFFI.CLRObjectMarshaler.GetProtoCoreType(param.ParameterType);
-                    formalParams.Add(new CLRFunctionEndPoint.ParamInfo() { CLRInfo = param, ProtoInfo = dsType });
-                }
-
-                CLRFunctionEndPoint fep = new CLRFunctionEndPoint(mInfo, formalParams);
-                
-                feps.Add(fep);
-            }
-
-            return feps;
-        }
```

Comment:
So this code has now basically been moved up to `FunctionLookup `at compile time?

Reply:
yes